multisig/tests/multisig_blackbox_test.rs

```diff
@@ -1050,3 +1154,1348 @@ fn multisig_non_board_member_sign_test() {
+        .world
+        .query()
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .amount_staked(RELAYER2_ADDRESS.to_managed_address())
+        .returns(ReturnsResult)
+        .run();
+
+    assert_eq!(remaining_stake_relayer2, BigUint::from(2_000u64));
+}
+
+#[test]
+fn test_propose_esdt_safe_set_current_transaction_batch_status_batch_empty() {
+    let mut state = MultiTransferTestState::new();
+
+    state.deploy_contracts_config();
+
+    let esdt_safe_address = ESDT_SAFE_ADDRESS;
+    let token_id = WEGLD_TOKEN_ID;
+    let amount = BigUint::<StaticApi>::from(1_000u64);
+
+    let statuses: MultiValueEncoded<StaticApi, TransactionStatus> =
+        MultiValueEncoded::from(ManagedVec::from_single_item(TransactionStatus::Pending));
+
+    state
+        .world
+        .tx()
+        .from(RELAYER1_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .propose_esdt_safe_set_current_transaction_batch_status(1u64, statuses)
+        .returns(ExpectError(4, "Current batch is empty"))
+        .run();
+}
+
+#[test]
+fn test_init_supply_from_child_contract_success() {
+    let mut state = MultiTransferTestState::new();
+
+    state.deploy_contracts_config();
+
+    let token_id = NATIVE_TOKEN_ID;
+    let amount = BigUint::from(1_000u64);
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .init_supply_from_child_contract(TokenIdentifier::from(token_id), amount.clone())
+        .single_esdt(&TokenIdentifier::from(token_id), 0, &amount.clone())
+        .run();
+
+    state
+        .world
+        .check_account(ESDT_SAFE_ADDRESS)
+        .esdt_balance(token_id, amount.clone());
+}
+
+#[test]
+fn test_add_unprocessed_refund_tx_to_batch_success() {
+    let mut state = MultiTransferTestState::new();
+
+    let mut board: MultiValueEncoded<StaticApi, ManagedAddress<StaticApi>> =
+        MultiValueEncoded::new();
+    board.push(ManagedAddress::from(RELAYER1_ADDRESS.eval_to_array()));
+    board.push(ManagedAddress::from(RELAYER2_ADDRESS.eval_to_array()));
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .init(
+            ESDT_SAFE_ADDRESS,
+            MOCK_MULTI_TRANSFER_ADDRESS,
+            BRIDGE_PROXY_ADDRESS,
+            BRIDGED_TOKENS_WRAPPER_ADDRESS,
+            PRICE_AGGREGATOR_ADDRESS,
+            1_000u64,
+            500u64,
+            2usize,
+            board,
+        )
+        .code(MULTISIG_CODE_PATH)
+        .new_address(MULTISIG_ADDRESS)
+        .run();
+
+    state.mock_multi_transfer_deploy();
+
+    let tx_id = 1u64;
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .add_unprocessed_refund_tx_to_batch(tx_id)
```

Comment:
check storages

Reply:
we do not have a view for the pending batches. what should i check instead?

Reply:
unprocessed_refund_txs(tx_id) has to be cleaned
last_batch_id before and last_batch_id after and should be + 1

Reply:
forgot to check earlier, but this endpoint calls the add_unprocessed_refund_tx_to_batch from a mock multi-transfer contract which does not set any storage. this endpoint will be tested from multitransfer directly